docgen/parser/parser.go

```diff
@@ -90,9 +92,51 @@ func parseMessages(pkgName string, comments map[string]*Comment, messages []*goo
 			Fields:  parseFields(p, comments, m.GetField()),
 		}
 	}
+
+	// once we've defined all Messages, populate the NestedMessages
+	// member of each message
+	for _, m := range res {
+		m.NestedMessages = populateNestedMessages(m, res)
+	}
 	return res
 }
 
+func populateNestedMessages(message *Message, messages map[string]*Message) []*Message {
```

Comment:
this needs a godoc - is the incoming map generally empty? what is it used for? why are we returning a slice instead of just telling the caller to use the map?